csharp/extractor/Semmle.Extraction.CSharp/Extractor/Extractor.cs

```diff
@@ -435,6 +435,8 @@ public static IEnumerable<Action> ReadSyntaxTrees(IEnumerable<string> sources, A
                 }
             }
 
+            syntaxTrees.Sort((a, b) => string.Compare(a.FilePath, b.FilePath, StringComparison.Ordinal));
```

Comment:
Should we instead respect the order in which the files were passed to the compiler in traced/binlog mode?

Reply:
The binlog extractor is not impacted by this change, because that uses the compilation from the complog API.

For traced extraction it would probably be better to use the order from the arguments, but I'd rather merge this PR and see if it solves our non-determinism issues, and do a follow up PR later. We're anyways using a random order currently.